Source/Dafny/Verifier/Translator.cs

```diff
@@ -3475,53 +3475,29 @@ public enum StmtType { NONE, ASSERT, ASSUME };
         builder.Add(TrAssumeCmd(f.tok, etran.TrExpr(en.E)));
       }
 
-      //generating assume J.F(ins) == C.F(ins)
```

Comment:
The diff here is deceptive: removing the `builder.Add(TrAssumeCmd(...))` line makes half of the lines here unused, but the other half is still needed if there is a result variable declared. Consider this as a two step edit: first removing those interspersed unneeded lines, and then moving the remainder inside the `if (resultVariable != null)` block.